crates/provider-wit-bindgen/tests/bindgen_kv_memory.rs

```diff
@@ -0,0 +1,113 @@
+use std::process::Stdio;
+
+use anyhow::{Context, Result};
+use escargot::CargoBuild;
+use tempfile::TempDir;
+use tokio::process::Command;
+use uuid::Uuid;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_kv_memory() -> Result<()> {
+    let wasmcloud_bin_path =
+        std::env::var("TEST_WASMCLOUD_BIN").context("no wasmcloud bin provided")?;
+    let wash_binary_path = std::env::var("TEST_WASH_BINARY").unwrap_or("wash".into());
+
+    // Build the wit-bindgen powered kv memory provider (~40s)
+    let provider_build_dir =
+        TempDir::new().context("failed to create temp dir for provider cargo build")?;
+    let provider_binary = CargoBuild::new()
+        .bin("kv-memory-bindgen")
+        .current_release()
+        .current_target()
+        .manifest_path("../../tests/providers/rust/kv-memory-bindgen/Cargo.toml")
+        .target_dir(provider_build_dir.path())
+        .run()
+        .context("failed to build provider binary")?;
+    let provider_binary_path = provider_binary
+        .path()
+        .canonicalize()
+        .context("failed to canonicalize")?
+        .display()
+        .to_string();
+
+    // Build the par file
+    // Run the provider from the rust host with wash
+    let provider_par_path = provider_build_dir
+        .path()
+        .join("kv-memory-bindgen.par.gz")
+        .display()
+        .to_string();
+    // wash par create --compress --name kv-memory-bindgen --capid wasi:keyvalue --binary path --destination provider.par.gz
+    Command::new(&wash_binary_path)
+        .args([
+            "par",
+            "create",
+            "--vendor",
+            "wasmcloud-test",
+            "--arch",
+            "x86_64-linux", // TODO: determine current platform w/ Rust
+            "--compress",
+            "--name",
+            "kv-memory-bindgen",
+            "--capid",
+            "wasi:keyvalue",
+            "--binary",
+            &provider_binary_path,
+            "--destination",
+            &provider_par_path,
+        ])
+        .status()
+        .await
+        .context("failed to create par gz")?;
+
+    // Start NATS
+    let nats_binary_path = std::env::var("TEST_NATS_BINARY").unwrap_or("nats-server".into());
```

Comment:
We already have a variable for this https://github.com/wasmCloud/wasmCloud/blob/28d2d6fc5e68ab8de12771fb3b0fb00617b32b30/tests/wasmbus.rs#L257

Reply:
The intent was not to influence the NATS version used by wasmcloud -- this is the NATS version used by the test -- not Wasmcloud proper -- note the `TEST_` prefix. 

All the other binaries that were introduced here have their ENV variables prefixed with `TEST_` so they can be changed without interfering with wasmcloud settings.

Reply:
> [EDIT] I see that `WASMCLOUD_NATS` is only used in test -- that should be prefixed with `TEST_` to avoid confusion with the actual ENV variables used by wasmcloud.

I am happy to figured that out. To answer your question - no, it should not be prefixed.